packages/cinterop/src/jvmCommon/kotlin/io/realm/interop/RealmInterop.kt

```diff
@@ -240,11 +240,11 @@ actual object RealmInterop {
 
     actual fun realm_object_add_notification_callback(obj: NativePointer, callback: Callback): NativePointer {
         return LongPointerWrapper(
-            realmc.realm_object_add_notification_callbackJNI(
+            realmc.register_object_notification_cb(
                 obj.cptr(),
-                object : io.realm.interop.NotificationCallback() {
+                object : NotificationCallback {
                     override fun onChange(pointer: Long) {
-                        callback.onChange(LongPointerWrapper(pointer))
+                        callback.onChange(LongPointerWrapper(pointer, managed = false))
```

Comment:
This will leak. Maybe we need to elaborate on why it is throwing, maybe an issue with the C-API, but if we really leave it as manage, I guess we need to release manually after the callback, or alternative find some other convention. I think there is copy/clone that I would suspect just increments the reference count if we need to construct managed pointers higher up the stack.